parachain/pallets/basic-channel/src/inbound/mod.rs

```diff
@@ -66,7 +66,8 @@ pub mod pallet {
 	pub type SourceChannel<T: Config> = StorageValue<_, H160, ValueQuery>;
 
 	#[pallet::storage]
-	pub type Nonce<T: Config> = StorageValue<_, u64, ValueQuery>;
+	// TODO: if eth addresses are user-generated, we need to use the blake2_128_concat hasher
```

Comment:
Actually, according to the doc below (see section "Real Accounts"), the eth addresses in this context wouldn't really be entirely user generated. Users would need to possess a valid private key that corresponds to user generated address in order to submit ethereum transactions which use the bridge.

https://www.shawntabrizi.com/substrate/transparent-keys-in-substrate/

So the suggestion would be to use the `Twox64Concat` hasher.

Reply:
@vgeddes Thanks! I've updated the hasher. If I'm understanding correctly, the eth address creation process is to create a key pair and use the last 20 bytes of `keccak(pubkey)` as the address. So we're relying on that section of a keccak hash being cryptographically secure. An attacker might be able to choose the pubkey and get hashes with a shared prefix from Twox64, but they still can't create storage keys with a shared prefix any easier than brute force because of keccak.

The pipeline I'm imagining for our storage key prefix is this:

pubkey -> keccak256 -> trunc to last 20 bytes -> twox64

Is keccak still secure wrt shared prefixes after being truncated? Haven't found an answer to this online, but seems like it would be a useful thing for Ethereum as a whole.